File: Person.cs
using System;

namespace Notebook
{
    public class Person
    {
        public int Id { get; set; }
        public int age { get; set; }
        public string name { get; set; }
        public string number { get; set; }
        public string email { get; set; }

        public Person() : this("not number")
        {
            //throw new Exception("Нельзя создать пустой контакт\n Необходим хотя бы номер");
        }
        public Person(string number) : this(number, "no name", 0)
        {

        }
        public Person(string number, string name) : this(number, name, 0)
        {

        }
        public Person(string number, string name, int age) : this(number, name, age, "example@mail.com")
        {

        }
        public Person(string number, string name, int age, string email)
        {
            this.age = age;
            this.name = name;
            this.number = number;
            this.email = email;
        }

        public void GetInfo()
        {
            Console.WriteLine(age);
            Console.WriteLine(name);
            Console.WriteLine(number);
            Console.WriteLine(email);
        }
    }
}

File: notebook_version_java/database/src/main/java/model/Person.java
package model;

import javax.persistence.Entity;
import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.GenerationType;

@Entity
@Table(name = "persons")
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(name = "name")
    private String name;
    @Column(name = "number")
    private String number;
    @Column(name = "email")
    private String email;
    @Column(name = "age")
    private int age;

    public Person(){}

    public Person(String name){
        this(name, "not_number");
    }

    public Person(String name, String number){
        this(name, number, "not_email");
    }

    public Person(String name, String number, String email){
        this(name, number, email, 0);
    }

    public Person(String name, String number, String email, int age){
        this.name = name;
        this.number = number;
        this.email = email;
        this.age = age;
    }
    public int getId(){return id;}

    public String getName(){
        return name;
    }

    public String getNumber(){
        return number;
    }

    public String getEmail(){
        return email;
    }

    public int getAge(){
        return age;
    }

    public void setName(String name){
        this.name = name;
    }

    public void setNumber(String number){
        this.number = number;
    }

    public void setEmail(String email){
        this.email = email;
    }

    public void setAge(int age){
        this.age = age;
    }
}